backend/commons/src/main/java/org/sonarsource/sonarlint/core/commons/GitBlameUtils.java

```diff
@@ -67,12 +71,14 @@ private static RepositoryBuilder getVerifiedRepositoryBuilder(Path basedir) {
     return builder;
   }
 
-  private static SonarLintBlameResult blameWithFilesGitCommand(Repository repo, Set<Path> gitRelativePath) {
+  private static SonarLintBlameResult blameWithFilesGitCommand(Repository repo, Set<Path> gitRelativePath, @Nullable UnaryOperator<String> fileContentProvider) {
```

Comment:
This is a minor nit-pick - I usually prefer having two separate constructors so we don't need to update the existing code by passing `nulls` (such as in the tests)